pkg/framework/machines.go

```diff
@@ -104,11 +104,13 @@ func GetMachineFromNode(client runtimeclient.Client, node *corev1.Node) (*mapiv1
 }
 
 // DeleteMachine deletes a specific machine and returns an error otherwise
-func DeleteMachine(client runtimeclient.Client, machine *mapiv1beta1.Machine) error {
+func DeleteMachine(client runtimeclient.Client, machines ...*mapiv1beta1.Machine) error {
```

Comment:
should this be `DeleteMachines` ?

Reply:
No

Reply:
Out of curiosity, how is it not? It gets a number of arguments?

Reply:
It revolves out of the fact you could pass a single instance. There is no reason for referencing it with DeleteMachine`s`

Reply:
If you are able to pass multiple machines, then it should be called `DeleteMachines`. If you want very much to have a function called `DeleteMachine` make a wrapper.

Reply:
I don't know... This also means that the comment to the function should be clarified. I would be stuck seeing one thing in a comment and other in a function.